rslib/src/scheduler/mod.rs

```diff
@@ -132,4 +141,36 @@ impl Collection {
         self.state.scheduler_info = None;
         self.storage.set_creation_stamp(stamp)
     }
+
+    /// Return forecast data for the next `days` days (capped at 7).
+    pub(crate) fn sched_forecast(
+        &mut self,
+        days: u32,
+    ) -> Result<Vec<anki_proto::scheduler::ReviewForecastDay>> {
+        use anki_proto::scheduler::ReviewForecastDay as PbDay;
+        let timing = self.timing_for_timestamp(TimestampSecs::now())?;
+        let today = timing.days_elapsed;
+        let mut out = Vec::new();
+        let want = days.min(7);
+
+        for offset in 0..want {
+            let target_day = today + offset;
+            let rev_cnt = self
+                .storage
+                .db
+                .prepare_cached("SELECT COUNT(*) FROM cards WHERE queue = 2 AND due = ?")
```

Comment:
For future reference, SQL should generally live in src/storage, with the collection method calling into the relevant storage function.